src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,10 +30,26 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
+        String[] breakAddress = addressSplitToGroups(address);
+        try {
+        	new Block(breakAddress[BLOCK]);
+        	new Street(breakAddress[STREET]);
+        	new Unit(breakAddress[UNIT]);
+        	new Postal(breakAddress[POSTAL]);
+        } catch (IllegalValueException e) {
+        	throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } catch (ArrayIndexOutOfBoundsException e) {
+        	throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
```

Comment:
Good attempt to use exceptions. However, note that exceptions should be used for exceptional situations. If invalid data entries is to be expected, we should not use exceptions to handle them.